### DISCUSSION TITLE
访问认证能否隐藏端口号？

### DISCUSSION THREAD

**Discussion by SherkeyXD** [natfrp/wiki#89] [想法]
如题，挂个端口号还是有点难受，尤其是我还开了一个verify的二级域名来专门做验证。
Expected: `verify.example.com:114514`  -> `verify.example.com`

**Comment by fakeboboliu** [comment]
这个想法很有趣，我会加入到我们的 TODO 中，但是可能不会得到很高的优先级，甚至可能进入 [TODO的终极结局](https://never.do)。
同时这个想法的具体实现也是有一些限制的，需要我们进行一些思考，原因如下：

 - 端口号没有 114514 的，可能要等到下一代 IP 标准扩充端口号字段才能实现了。
 - 客户端尤其是启动器的模型限制，多隧道之间互相影响相对比较麻烦，我们需要将客户端对外影响的能力限制到最小来保证安全。
 - 完善此功能可能需要服务器端的功能。访问认证的主要设计都是为了保护安全和隐私，我们想要尽可能把选择权交到用户手中。因此我们需要警惕和最小化服务器的介入量和介入权限。我们选择了一些相对比较拧巴的形式来处理需要与服务器交互的方便化的功能，如可选的通过后台认证，继续进行这样的设计需要一些思考和讨论拧巴出来一个符合我们原则的结果。
 - 如果你 [愿用隐私换效率](https://www.zhihu.com/question/269959475)，上一条当我没说。
 
 与此同时，我建议如果确有比较迫切的需求，可以自己开发一个小工具配合 https 隧道来做，这并不难。

**Comment by SanaeAkahane** [comment]
HTTPS 隧道本地 IP 设置成节点 IP (雾)
并把 HTTPS 隧道本地端口 设为 TCP 访问认证隧道的 远程端口（（（

没有实际试验，就当说个笑话罢
